airbyte-integrations/connectors/source-square/source_square/source.py

```diff
@@ -367,12 +363,18 @@ def stream_slices(self, **kwargs) -> Iterable[Optional[Mapping[str, Any]]]:
                 "No locations found. Orders cannot be extracted without locations. "
                 "Check https://developer.squareup.com/explorer/square/locations-api/list-locations"
             )
-            exit(1)
+            return [None]
 
         separated_locations = separate_items_by_count(location_ids, self.locations_per_requets)
         for location in separated_locations:
             yield {"location_ids": location}
 
+    def read_records(self, stream_slice: Mapping[str, Any] = None, **kwargs) -> Iterable[Mapping[str, Any]]:
```

Comment:
it will be not needed if we return empty list for streamslice

Reply:
Yes, thanks. For some reason, I thought we need to process the `stream_slices` states in `read_records`. Done